File: Python/Easy/Node Depths/solution_1.py
#Iterative Approach

def nodeDepths(root):
    depths = 0
    stack = [{"node": root , "depth":0}]

    while len(stack) > 0:
        nodeInfo = stack.pop()
        node, depth = nodeInfo["node"], nodeInfo["depth"]

        if node is None:
            continue

        depths += depth
        stack.append({"node":node.left , "depth":depth + 1})
        stack.append({"node":node.right , "depth":depth + 1})

    return depths
        

# This is the class of the input binary tree.
class BinaryTree:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

File: Go/Easy/Node Depths/solution_1.go
package main

type BinaryTree struct {
	Value       int
	Left, Right *BinaryTree
}

type Level struct {
	Root  *BinaryTree
	Depth int
}

func NodeDepths(root *BinaryTree) int {
	depths := 0
	stack := []Level{{Root: root, Depth: 0}}

	var top Level

	for len(stack) > 0 {

		top, stack = stack[len(stack)-1], stack[:len(stack)-1]
		node, depth := top.Root, top.Depth

		if node == nil {
			continue
		}
		depths += depth
		stack = append(stack, Level{Root: node.Left, Depth: depth + 1})
		stack = append(stack, Level{Root: node.Right, Depth: depth + 1})
	}
	return depths
}
